### PR TITLE
Automated cherry pick of #114093: use etcd 3.5.6-0 after promotion

### PR DESCRIPTION
**Pull request by pacoxu**
Cherry pick of #114093 on release-1.25.

#114093: use etcd 3.5.6-0 after promotion

For details on the cherry pick process, see the [cherry pick requests](https://git.k8s.io/community/contributors/devel/sig-release/cherry-picks.md) page.

```release-note

```

### CODE REVIEW

**Comment by neolit123** [comment]
i am +1 to update etcd across the branches. problem is that historically these PRs are not accepted, even if the etcd maintainers recommend the update.

**Comment by xmudrii** [comment]
@neolit123 We recently updated all release branches to 3.5.5 (#113197, #113099, #113100) so I don't see any problem with bumping it to 3.5.6 (at least from the Release Engineering side). That being said, I'm also +1 for accepting those cherry-picks.

**Comment by aojea** [comment]
this sounds like a nasty bug https://groups.google.com/a/kubernetes.io/g/dev/c/sEVopPxKPDo/m/9ME3CzicBwAJ that justify the backports
/assign @liggitt @dims 

for deps approval

**Comment by saschagrunert** [comment]
Tomorrow is the cherry-pick deadline and not all backports of this series have been `lgtm`ed and `approved`.

**Comment by pacoxu** [comment]
@kubernetes/release-managers

**Review by xmudrii — APPROVED** [review]
/lgtm
/approve

**Pull request merged**
